### PR TITLE
[TASK] Change rector configuration and run

### PR DESCRIPTION
**Pull request by extcode**
I added two rector rules to skip list, because they change the order of constructor arguments or replace injector methods in the abstract ActionController class.

Fixes: #530

### CODE REVIEW

File: Classes/Controller/Backend/Order/DocumentController.php [hunk 1]
```diff
@@ -26,19 +26,10 @@
...
-    }
-
-    public function injectItemRepository(ItemRepository $itemRepository): void
-    {
-        $this->itemRepository = $itemRepository;
-    }
+    public function __construct(
+        protected readonly PersistenceManager $persistenceManager,
+        protected readonly ItemRepository $itemRepository
+    ) {}
```
Comment (rintisch) [inline Classes/Controller/Backend/Order/DocumentController.php:32]: I understood that you do not want to use DI via `construct()` but now all dependencies are injected that way!? (Same for all other controller)

File: Classes/Controller/Cart/ActionController.php [hunk 2]
```diff
@@ -52,15 +52,7 @@ public function injectPaymentMethodsService(PaymentMethodsServiceInterface $paym
...
 
-    public function injectShippingMethodsService(ShippingMethodsServiceInterface $shippingMethodsService): void
-    {
-        $this->shippingMethodsService = $shippingMethodsService;
-    }
-
-    public function injectSpecialOptionsService(SpecialOptionsServiceInterface $specialOptionsService): void
-    {
-        $this->specialOptionsService = $specialOptionsService;
-    }
```
Comment (rintisch) [inline Classes/Controller/Cart/ActionController.php:63]: Why are `specialOptions` no longer injected here?

File: Classes/Domain/Model/Order/ProductAdditional.php [hunk 3]
```diff
@@ -14,27 +14,9 @@
...
-        string $key,
-        string $value,
-        string $data = ''
-    ) {
-        $this->additionalType = $type;
-        $this->additionalKey = $key;
-        $this->additionalValue = $value;
-        $this->additionalData = $data;
-    }
+    public function __construct(protected string $additionalType, protected string $additionalKey, protected string $additionalValue, protected string $additionalData = '') {}
```
Comment (rintisch) [inline Classes/Domain/Model/Order/ProductAdditional.php:19]: Might be better to split it on multiple lines?

File: Classes/Domain/Model/Order/Tax.php [hunk 4]
```diff
@@ -16,19 +16,9 @@
...
-    public function __construct(
-        float $tax,
-        TaxClass $taxClass
-    ) {
-        $this->tax = $tax;
-        $this->taxClass = $taxClass;
-    }
+    public function __construct(#[Validate(['validator' => 'NotEmpty'])]
+        protected float $tax, #[Validate(['validator' => 'NotEmpty'])]
+        protected TaxClass $taxClass) {}
```
Comment (rintisch) [inline Classes/Domain/Model/Order/Tax.php:21]: Maybe the attributes should be in the same line as the argument they belong to?

**Review by rintisch — COMMENTED** [review]
I added some comments :)

File: Classes/Controller/Backend/Order/DocumentController.php (hunk 1 above)
Comment (extcode) [reply Classes/Controller/Backend/Order/DocumentController.php:32]: I think we have misunderstood each other. I have nothing against dependency injection via the constructor method, but in abstract classes it's not so nice. I think if someone extends the DocumentController, for example, then it's quite OK for me that the Constructor arguments have to be passed via a parent::__construct().
This is still possible. I had already considered declaring all the controllers as final.

Comment (rintisch) [reply Classes/Controller/Backend/Order/DocumentController.php:32]: Okay!

File: Classes/Controller/Cart/ActionController.php (hunk 2 above)
Comment (extcode) [reply Classes/Controller/Cart/ActionController.php]: I have probably not restored everything here that rector had changed. I fixed it with 1b75dec.

Comment (rintisch) [reply Classes/Controller/Cart/ActionController.php]: Great!

File: Classes/Domain/Model/Order/ProductAdditional.php (hunk 3 above)
Comment (extcode) [reply Classes/Domain/Model/Order/ProductAdditional.php]: Fix this with e91a033.

File: Classes/Domain/Model/Order/Tax.php (hunk 4 above)
Comment (extcode) [reply Classes/Domain/Model/Order/Tax.php]: Fix this with e91a033.

**Pull request merged**